### PR TITLE
Package Mantid Workbench for OSX

### PR DESCRIPTION
**Pull request by samueljackson92**
**Description of work.**

This adds an application bundle for the workbench for OSX.

**To test:**

This needs to be tested on all supported platforms incase we have broken the packaging process for another system as we've added these changes.

 - [x] Tested on Linux
 - [x] Tested on OSX
 - [x] Tested on Window

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by samueljackson92** [comment]
@keeeto The time has come. Would you kindly give this a test drive on your mac?

**Comment by keeeto** [comment]
* Add `"libboost_python27-mt.dylib"` to the `make_package.rb`
* Allow warnings when importing `PyQt5`
* Ensure that `mantid-scripts.pth` is in the Workbench package

**Comment by samueljackson92** [comment]
@keeeto Should be ready for another test when you can.

**Review by martyngigg — CHANGES_REQUESTED** [review]
Unfortunately we have broken something with the workbench packaging on Linux:
```
$ ls /opt/mantidnightly/bin                    
ipython_widget        mantidplot      mantidplot.py    mantidplotrc.pyc   mantidqtpython.so              mantid_reset_settings.sh  paraview-config  pvpython        pymantidplot
launch_mantidplot.sh  MantidPlot      mantidplot.pyc   Mantid.properties  mantid_qt_settings_editor.py   mantid-scripts.pth        pvbatch          pvrenderserver  workbench
mantid                MantidPlot_exe  mantidplotrc.py  mantidpython       mantid_qt_settings_editor.pyc  mantidworkbench           pvdataserver     pvserver
$ ls /opt/mantidunstable/bin
ipython_widget        lib         MantidPlot      mantidplot.pyc    Mantid.properties  mantid_qt_settings_editor.py   mantid-scripts.pth  pvbatch       pvrenderserver
launch_mantidplot.sh  mantid      MantidPlot_exe  mantidplotrc.py   mantidpython       mantid_qt_settings_editor.pyc  mantidworkbench     pvdataserver  pvserver
launch_workbench      mantidplot  mantidplot.py   mantidplotrc.pyc  mantidqtpython.so  mantid_reset_settings.sh       paraview-config     pvpython      pymantidplot
$ ls /opt/mantidunstable/lib    
libMantidAPI.so          libMantidHistogramData.so  libMantidParallel.so             libMantidQtWidgetsInstrumentViewQt4.so   libMantidQtWidgetsSliceViewerQt4.so     mantidqt-3.13.20190108.908.egg-info
libMantidBeamline.so     libMantidIndexing.so       libMantidPythonInterfaceCore.so  libMantidQtWidgetsInstrumentViewQt5.so   libMantidQtWidgetsSpectrumViewerQt4.so  MantidWorkbench-3.13.20190108.908.egg-info
libMantidCatalog.so      libMantidKernel.so         libMantidQtWidgetsCommonQt4.so   libMantidQtWidgetsLegacyQwtQt4.so        libMantidTypes.so                       paraview-5.4
libMantidDataObjects.so  libMantidNexusGeometry.so  libMantidQtWidgetsCommonQt5.so   libMantidQtWidgetsMplCppQt5.so           libMantidVatesAPI.so                    workbench
libMantidGeometry.so     libMantidNexus.so          libMantidQtWidgetsFactoryQt4.so  libMantidQtWidgetsRefDetectorViewQt4.so  mantidqt
```

It seems that we have put the `lib` folder in `bin` but this shouldn't be there.

File: buildconfig/CMake/PythonPackageTargetFunctions.cmake [hunk 1]
```diff
@@ -93,8 +94,22 @@ CustomInstallLib = patch_setuptools_command('install_lib')
         # On UNIX systems install the workbench executable directly.
         # The Windows case is handled with a custom startup script installed in WindowsNSIS
         if ( NOT WIN32 )
-          install(PROGRAMS ${_setup_py_build_root}/install/bin/${_executable_name} DESTINATION bin)
+            install(PROGRAMS ${_setup_py_build_root}/install/bin/${pkg_name}
+                DESTINATION ${WORKBENCH_BIN_DIR}
+                RENAME launch_workbench)
         endif()
+
+    # register the "installed" components with cmake so it will carry them over
```
Comment (martyngigg) [inline buildconfig/CMake/PythonPackageTargetFunctions.cmake]: I think this is the cause of the duplicate `lib` directory installation. Is this not a duplicate of line `87`?

Comment (martyngigg) [reply buildconfig/CMake/PythonPackageTargetFunctions.cmake]: I think line `87` is the one we need to keep.

**Comment by samueljackson92** [comment]
@keeeto When you have time could you retest this on OSX. I made some changes to the DMG installer that it would be worth just checking on someone else's machine.

@martyngigg When you have time I think this is good to be retested on Linux and tested on Windows.

File: scripts/ErrorReporter/error_dialog_app.py [hunk 2]
```diff
@@ -15,16 +15,17 @@
 command_line_args = parser.parse_args()
 
 sys.path.insert(0, command_line_args.directory)
+print(sys.path)
```
Comment (martyngigg) [inline scripts/ErrorReporter/error_dialog_app.py]: We left a print statement in...

File: buildconfig/CMake/PythonPackageTargetFunctions.cmake [hunk 3]
```diff
@@ -93,8 +94,17 @@ CustomInstallLib = patch_setuptools_command('install_lib')
         # On UNIX systems install the workbench executable directly.
         # The Windows case is handled with a custom startup script installed in WindowsNSIS
         if ( NOT WIN32 )
-          install(PROGRAMS ${_setup_py_build_root}/install/bin/${_executable_name} DESTINATION bin)
+            install(PROGRAMS ${_setup_py_build_root}/install/bin/${pkg_name}
+                DESTINATION ${WORKBENCH_BIN_DIR}
+                RENAME launch_workbench)
         endif()
+
+    # install the generated executable - only tested with "workbench"
```
Comment (martyngigg) [inline buildconfig/CMake/PythonPackageTargetFunctions.cmake]: I think we've duplicated these lines so lines 102-107 can be removed.

File: buildconfig/CMake/PythonPackageTargetFunctions.cmake [hunk 4]
```diff
@@ -93,8 +94,17 @@ CustomInstallLib = patch_setuptools_command('install_lib')
         # On UNIX systems install the workbench executable directly.
         # The Windows case is handled with a custom startup script installed in WindowsNSIS
         if ( NOT WIN32 )
-          install(PROGRAMS ${_setup_py_build_root}/install/bin/${_executable_name} DESTINATION bin)
+            install(PROGRAMS ${_setup_py_build_root}/install/bin/${pkg_name}
+                DESTINATION ${WORKBENCH_BIN_DIR}
+                RENAME launch_workbench)
```
Comment (martyngigg) [inline buildconfig/CMake/PythonPackageTargetFunctions.cmake]: The renaming here has broken the Linux packaging.  Let's remove the `RENAME launch_workbench` and leave it as workbench. We'll then need to update [MantidWorkbench_osx_launcher](https://github.com/samueljackson92/mantid/blob/workbench-package-osx/buildconfig/CMake/Packaging/osx/MantidWorkbench_osx_launcher#L6) to change `./launch_workbench` to just `./workbench`.

**Review by martyngigg — CHANGES_REQUESTED** [review]

**Review by martyngigg — DISMISSED** [review]
Works for me now!

**Comment by martyngigg** [comment]
Gatekeepers: Please wait for confirmation from tests on the other OSs

**Comment by DTasev** [comment]
I will try this change on Windows

**Comment by DTasev** [comment]
Package built, installed and ran MantidPlot and Workbench successfully. No issues found

**Review by dtasev — DISMISSED** [review]

**Comment by samueljackson92** [comment]
Just waiting on a retest of OSX now. One last time @keeeto?

**Review by martyngigg — CHANGES_REQUESTED** [review]
The `MantidWorkbench` bundle is missing `sip.so` and `skimage` from the `Contents/MacOS` directory

**Review by martyngigg — APPROVED** [review]
It works!

**Pull request merged**